src/gui/qgsmapcanvas.cpp

```diff
@@ -778,8 +796,26 @@ QgsRectangle QgsMapCanvas::imageRect( const QImage& img, const QgsMapSettings& m
 
 void QgsMapCanvas::mapUpdateTimeout()
 {
-  const QImage& img = mJob->renderedImage();
-  mMap->setContent( img, imageRect( img, mJob->mapSettings() ) );
+  QImage img = mJob->renderedImage();
+  magnifyImage( img );
+
+  mMap->setContent( img, imageRect( img, mSettings ) );
+}
+
+void QgsMapCanvas::magnifyImage( QImage &img )
+{
+  if ( mMagnificationLevel != 1.0 )
+  {
+    img = img.scaled( img.width() * mMagnificationLevel,
```

Comment:
Hmmm.. I disagree. I see the main use for a magnifier as zooming in to see more detail, which a pixelated view won't give. A non pixelated zoom would also help with cartographic layouts, since you could then zoom in to finely adjust label placement for high resolution outputs. If this feature is solely about a temporary pixelated zoom in, then I think it would be better implemented as a zoomed rectangle map tool or some other cursor-following approach.